daemon/cmd/daemon_main.go

```diff
@@ -1338,6 +1341,21 @@ func initEnv(cmd *cobra.Command) {
 		option.Config.NeedsRelaxVerifier = true
 	}
 
+	if option.Config.LocalRouterIP != "" {
+		if option.Config.IPAM != "" {
+			log.Fatalf("Cannot specify %s along with %s, leave router IP unspecified if Cilium is handling IPAM.", option.LocalRouterIP, option.IPAM)
+		}
+		if option.Config.Tunnel != option.TunnelDisabled {
+			log.Fatalf("Cannot specify %s in tunnel mode.", option.LocalRouterIP)
+		}
+		if !option.Config.EnableEndpointRoutes {
+			log.Fatalf("Cannot specify %s without %s.", option.LocalRouterIP, option.EnableEndpointRoutes)
```

Comment:
One of these days, I'll submit a PR to disallow this form of logging (format variations) in favour of structured logging. That said, for Fatal messages there's not as strong of a concern around structured logging since the error will be very loud with the cilium pods failing on startup and you won't need to correlate the fields with other logs elsewhere in your system.

Reply:
^ Apparently this is already in the linter, at least for Info() messages.